lib/std/ascii.zig

```diff
@@ -47,214 +51,171 @@ pub const control_code = struct {
-pub fn isAlNum(c: u8) bool {
-    return (combinedTable[c] & ((@as(u8, 1) << @enumToInt(tIndex.Alpha)) |
-        @as(u8, 1) << @enumToInt(tIndex.Digit))) != 0;
-}
+        return .{ .table = table };
+    }
+
+    fn contains(self: CombinedTable, char: u8, index: Index) bool {
+        return (self.table[char] & (@as(u8, 1) << @enumToInt(index))) != 0;
+    }
+};
+
+/// The combined table for fast lookup.
+///
+/// This is not used in `ReleaseSmall` to save 256 bytes at the cost of
+/// a small decrease in performance.
+const combined_table: ?CombinedTable = if (@import("builtin").mode == .ReleaseSmall)
+    null
+else
+    CombinedTable.init();
 
-pub fn isAlpha(c: u8) bool {
-    return inTable(c, tIndex.Alpha);
+/// Returns `true` if the character is a control character.
+///
+/// See also: `control`
+pub fn isControl(char: u8) bool {
+    if (combined_table) |table|
+        return table.contains(char, .control)
+    else
+        return isControlNaive(char);
 }
 
-pub fn isCntrl(c: u8) bool {
-    return c < 0x20 or c == 127; //DEL
+/// Returns `true` if the character is alphanumeric. This is case-insensitive.
+pub fn isAlphanumeric(char: u8) bool {
+    if (combined_table) |table|
+        return table.contains(char, .alphanumeric)
+    else
+        return isAlphanumericNaive(char);
 }
 
-pub fn isDigit(c: u8) bool {
-    return inTable(c, tIndex.Digit);
+/// Returns `true` if the character is alphabetic. This is case-insensitive.
+pub fn isAlphabetic(char: u8) bool {
+    if (combined_table) |table|
+        return table.contains(char, .alphabetic)
+    else
+        return isAlphabeticNaive(char);
 }
 
-pub fn isGraph(c: u8) bool {
-    return inTable(c, tIndex.Graph);
+pub fn isDigit(char: u8) bool {
+    return char >= '0' and char <= '9';
 }
 
-pub fn isLower(c: u8) bool {
-    return inTable(c, tIndex.Lower);
+/// Returns `true` if the character has some graphical representation and can be printed.
+pub fn isPrintable(char: u8) bool {
+    return char >= ' ' and char <= '~';
 }
 
-pub fn isPrint(c: u8) bool {
-    return inTable(c, tIndex.Graph) or c == ' ';
+pub fn isLower(char: u8) bool {
+    return char >= 'a' and char <= 'z';
 }
 
-pub fn isPunct(c: u8) bool {
-    return inTable(c, tIndex.Punct);
+pub fn isUpper(char: u8) bool {
+    return char >= 'A' and char <= 'Z';
 }
 
-pub fn isSpace(c: u8) bool {
-    return inTable(c, tIndex.Space);
+pub fn isWhitespace(char: u8) bool {
+    if (combined_table) |table|
+        return table.contains(char, .whitespace)
+    else
+        return isWhitespaceNaive(char);
 }
 
-/// All the values for which isSpace() returns true. This may be used with
-/// e.g. std.mem.trim() to trim whiteSpace.
-pub const spaces = [_]u8{ ' ', '\t', '\n', '\r', control_code.VT, control_code.FF };
+/// All the values for which `isWhitespace()` returns `true`.
+/// This may be used with e.g. `std.mem.trim()` to trim whitespace.
+pub const whitespace = [_]u8{ ' ', '\t', '\n', '\r', control.VT, control.FF };
```

Comment:
s/whitespace/whitespaces

Reply:
Whitespace is a valid plural for whitespace (it's an uncountable noun).